src/api/providers/kilocode-openrouter.ts

```diff
@@ -59,21 +59,7 @@ export class KilocodeOpenrouterHandler extends OpenRouterHandler {
 
 	override getModel() {
 		let id = this.options.kilocodeModel ?? this.defaultModel
-		let info = this.models[id]
-		let defaultTemperature = 0
-
-		if (!info) {
-			const defaultInfo = this.models[this.defaultModel]
-			if (defaultInfo) {
-				console.warn(`${id} no longer exists, falling back to ${this.defaultModel}`)
-				id = this.defaultModel
-				info = defaultInfo
-			} else {
-				console.warn(`${id} no longer exists, falling back to ${openRouterDefaultModelId}`)
-				id = openRouterDefaultModelId
-				info = openRouterDefaultModelInfo
-			}
-		}
+		let info = this.models[id] ?? openRouterDefaultModelInfo
```

Comment:
this may be correct, but it reads weird when not knowing the underlying semantics anyhow:

 - we pick an id from options and fall back to a defaultModel (presumably id?)
 - then we try to resolve that id, but fall back to openRouterDefaultModelInfo
  - then later on, we use both id and info as if they're in sync.
  
 At the very least that's rather clever; the code doesn't itself converge on a consistent state; if it does so that's because non-locally stuff is in sync.

Reply:
(which I suspect it is, but if ever we were to accidentally have different defaults, then this is bound to do weird things)

Reply:
In general, it feels weird to be passing around both id and info when info contains id (but that seems to be the pattern?)

Reply:
> then we try to resolve that id, but fall back to openRouterDefaultModelInfo

this is not correct (I mean the code is not correct), but there is a race condition between `getModel` and `fetchModel` somewhere that we haven't been able to reproduce, but we get complaints if we just throw an error here.

> In general, it feels weird to be passing around both id and info when info contains id (but that seems to be the pattern?)

yeah, it's on the long list of things to improve